pkg/volume/csi/csi_mounter.go

```diff
@@ -367,6 +367,10 @@ func (c *csiMountMgr) TearDownAt(dir string) error {
 
 	// clean mount point dir
 	if err := removeMountDir(c.plugin, dir); err != nil {
+		if isCorruptedDir(dir) {
+			klog.Warningf(log("dir[%s] is corrupted, error: %v, skip mount dir removal", dir, err))
+			return nil
+		}
 		return errors.New(log("mounter.TearDownAt failed to clean mount dir [%s]: %v", dir, err))
```

Comment:
In other places - if we can't cleanup the mountDir because of `stat` failures - then we don't usually finish teardown process and we are in fact letting the pod hang on purpose. 

I think at very minimum this would result in - orphan pod directories left over on the node.

Reply:
that's original behavior if SMB server is down, and pod would be be terminating state forever, shall we fix this issue or mark as by design?

Reply:
Yeah I am not sure. We do same thing (i.e leaving the pod hanging forever) for NFS too. I feel like we should do better job and somehow standardize around this. This also means that - code that touches unmount should have some safety built-in.

Reply:
@drigz @andyzhangx the question is - do we even have to change this code at all? This code affects all CSI drivers and as per CSI spec - the removal of mount point directory should be done by the driver. I know traditionally before we fixed it, it was the kubelet which was creating the node-publish path, but it was a bug.

So after we fix that - shouldn't the CSI driver remove the publish path on `NodeUnpublish` and then just leave the cleanup of parent directories to the kubelet? And in which case - kubelet should have no trouble removing the path because, if `NodeUnpublish` succeeded then- kubelet *should* be able to remove the directory.

Reply:
see the CSI spec - https://github.com/container-storage-interface/spec/blob/master/spec.md#nodeunpublishvolume 

```
 The SP MUST delete the file or directory it created at this path.
  // This is a REQUIRED field.
```

Reply:
so with https://github.com/kubernetes/kubernetes/issues/101332, we should remove Line 369 in this PR?

Reply:
that call also removes the parent directory, but removal of `dir` by the kubelet is not necessary and should be done by the driver. Having said that - I took a look at existing drivers and I think most of them don't remove the publish_path correctly on `NodeUnpublish` - so we can't remove that code right away but we have to capture this via release notes and give time to users.